src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Telemetry/TelemetryOnceEmitter.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading;
+using NuGet.Common;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    /// <summary>
+    /// Ensures a given telemetry event is emitted once. Useful for counterfactual logging.
```

Comment:
```suggestion
    /// Ensures a given telemetry event is emitted once.
```

I mentioned this in another comment, but I don't think we should add an implication that counterfactual logging should be limited to only logging a single telemetry event.